### DISCUSSION TITLE
`tar_make()` colors other function's outputs red

### DISCUSSION THREAD

**Discussion by pwseo** [ropensci/targets#811] [Help] [answered]
When running a pipeline via `tar_make()`, one usually gets the default list of targets and their status. This list is usually red colored.
However, in some cases, some targets may need to write output to stdout, in which case it is also (wrongly, in my opinion) colored red.
This happens even when setting `reporter = "silent"` when invoking `tar_make()`. Is this a bug, or is there a rationale for this behaviour of which I'm not aware?
(there may be some way to overcome this issue, but I did not manage to do it, at least not yet; if someone knows how, I would love to know)

**Comment by wlandau** [comment, marked as answer]
targets does not try to interfere with ordinary stdout, and the red text comes from how tar_make() runs the pipeline in a callr process.

**Reply by pwseo** [reply]
Yes, I know targets does not modify the functions' output, but is there a way to prevent it from coloring subsequent output red? (or at least change the color)

**Reply by wlandau** [reply]
You could use the cli package to enforce specific colors. Maybe callr has something that you could use through the callr_arguments argument of tar_make()

**Reply by pwseo** [reply]
Oh, I see. I thought it was `targets` that was colorizing its output. So, if i understand correcly, the red color is an artifact from either `callr` or `processx`, the packages used under the hood to invoke each pipeline's execution.
I'm sorry for the mix up.
